crates/protocol/src/batch/validity.rs

```diff
@@ -0,0 +1,53 @@
+//! Contains the [BatchValidity] and its encodings.
+
+/// Batch Validity
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum BatchValidity {
+    /// The batch is invalid now and in the future, unless we reorg
+    Drop,
+    /// The batch is valid and should be processed
+    Accept,
+    /// We are lacking L1 information until we can proceed batch filtering
+    Undecided,
+    /// The batch may be valid, but cannot be processed yet and should be checked again later
+    Future,
+    /// Introduced in Holocene, a special variant of the Drop variant that signals not to flush
+    /// the active batch and channel, in the case of processing an old batch
+    Past,
+}
+
+impl BatchValidity {
+    /// Returns if the batch is accepted.
```

Comment:
nit: it always returns

```suggestion
    /// Returns whether the batch is accepted.
```